posts/api/views.py

```diff
@@ -0,0 +1,48 @@
+from rest_framework.generics import CreateAPIView, DestroyAPIView, ListAPIView, RetrieveAPIView, RetrieveUpdateAPIView
+from rest_framework.permissions import AllowAny, IsAuthenticated, IsAuthenticatedOrReadOnly
+import jwt
+
+from posts.models import Post
+from .serializers import PostCreateSerializer, PostDetailSerializer, PostListSerializer, PostUpdateSerializer
+
+
+def jwt_decoder(encoded_token):
+    return jwt.decode(encoded_token, secret_base_encoded=False, algorithms=['HS256'])
+
+
+class PostCreateAPIView(CreateAPIView):
+    queryset = Post.objects.all()
+    serializer_class = PostCreateSerializer
+    permission_classes = (IsAuthenticated,)
+
+    def perform_create(self, serializer):
+        payload = jwt_decoder(self.request.headers['Token'])
```

Comment:
Getting an error on this line. I guess `self.request.headers['Token']` is incorrect. It should be taken from corresponding `Authorization` header.

Reply:
The `Authorization` header might require `split` as it has _Bearer_ as prefix to it

Reply:
You will have to pass two keys in the header, 
1. Authorization: Normal access token
2. token: The access token without a base 64 encoded secret

Reply:
Okay, so how that secret is getting verified with the backend actually?

Reply:
Like, there has to some `secret` fixed on the backend too, how are you planning to do that?

Reply:
I have fixed this in the newer commits!